test/org/sagebionetworks/bridge/validators/SubpopulationValidatorTest.java

```diff
@@ -0,0 +1,51 @@
+package org.sagebionetworks.bridge.validators;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import org.sagebionetworks.bridge.exceptions.InvalidEntityException;
+import org.sagebionetworks.bridge.models.studies.Subpopulation;
+
+import com.google.common.collect.Sets;
+
+public class SubpopulationValidatorTest {
```

Comment:
other unit tests?
* valid subpop with all params?
* valid subpop with no params?
* edge case: valid subpop with maxAppVersion equal to minAppVersion
* maxAppVersion less than minAppVersion
* null allOfGroup
* null noneOfGroup
* allOfGroup contains wrong group